action.yml

```diff
@@ -0,0 +1,124 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: SSH to Google Cloud Platform compute instances
+author: Google LLC
+description: |-
+  Use this action to connect to Google Cloud Platform instances via ssh and execute user’s commands.
+
+inputs:
+  instance_name:
+    description: |-
+      Name of the virtual machine instance to SSH into.
+    required: true
+
+  zone:
+    description: |-
+      Zone of the instance to connect to.
+    required: true
+
+  user:
+    description: |-
+      Username with which to SSH. If omitted, the user login name is used.
+      GH action runner uses 'runner' as a default name.
+      If using OS Login, USER will be replaced by the OS Login user.
+    required: false
+
+  container:
+    description: |-
+      The name or ID of a container inside of the virtual machine instance to connect to.
+      This only applies to virtual machines that are using a Google Container-Optimized 
+      virtual machine image.
+    required: false
+
+  ssh_key_file_path:
+    description: |-
+      Absolute path to the SSH key file. By default, this is '/home/runner/.ssh/google_compute_engine'.
+    required: false
+
+  ssh_key_expire_after:
+    description: |-
+      The maximum length of time an SSH key is valid for once created and installed, e.g. 2m for 2 minutes.
+    required: false
+
+  ssh_args:
+    description: |-
+      Additional flags to be passed to ssh tool. Example: '-vvv -L 80:%INSTANCE%:80'.
+    required: false
+
+  internal_ip:
+    description: |-
+      Connect to instances using their internal IP addresses rather than their external IP addresses.
+      Use this to connect from one instance to another on the same VPC network, over a VPN connection,
+      or between two peered VPC networks.
+    required: false
+
+  tunnel_through_iap:
+    description: |-
+      Tunnel the ssh connection through Cloud Identity-Aware Proxy for TCP forwarding.
+    required: false
+
+  command:
+    description: |-
+      A command to run on the virtual machine.
+      Action runs the command on the target instance and then exits.
+    required: false
+
+  entrypoint:
+    description: |-
+      A path for bash script to run on the virtual machine.
```

Comment:
Why bash only?

Reply:
@sethvargo 
The original idea was to allow users create some "script.sh" file and then we automatically do: `bash -c \"<script.sh content>\". 

What's your idea here?

Reply:
On Windows, it's be Powershell I think